x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineFilters.java

```diff
@@ -135,6 +132,32 @@ private static ScopedFilter scopeFilter(List<Expression> filters, LogicalPlan le
         return new ScopedFilter(rest, leftFilters, rightFilters);
     }
 
+    // split the filter condition in 2 parts:
+    // 1. filters that can be applied only to the right
+    // 2. filters that can be applied to both sides
+    // 3. filters that can be applied only to the left
```

Comment:
I'd replace the "can be applied" with "reference" and then at the end "attributes": "1. filters that reference only to the RHS attributes" (or "..the right attributes"), since we're not applying those as such.